userspace/libsinsp/parsers.cpp

```diff
@@ -4525,15 +4524,13 @@ void sinsp_parser::parse_setregid_exit(sinsp_evt &evt) const {
 }
 
 void sinsp_parser::parse_setuid_exit(sinsp_evt &evt) const {
-	sinsp_evt *enter_evt = &m_tmp_evt;
-
 	//
 	// Extract the return value
 	//
 	const int64_t retval = evt.get_syscall_return_value();
 
-	if(retval == 0 && retrieve_enter_event(*enter_evt, evt)) {
-		uint32_t new_euid = enter_evt->get_param(0)->as<uint32_t>();
+	if(retval == 0) {
```

Comment:
I guess here we should check that the event has the "new format", this means checking that the number of parameter is greater than 1, before accessing the uid:
```suggestion
	if(retval == 0 && evt.get_num_params() > 1) {
```